src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectKNuGetProject.cs

```diff
@@ -60,19 +61,20 @@ public override Task PreProcessAsync(INuGetProjectContext nuGetProjectContext, C
             return base.PreProcessAsync(nuGetProjectContext, token);
         }
 
-        [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD002", Justification = "NuGet/Home#4833 Baseline")]
         public ProjectKNuGetProject(INuGetPackageManager project, string projectName, string uniqueName, string projectId)
         {
             _project = project;
             InternalMetadata.Add(NuGetProjectMetadataKeys.Name, projectName);
             InternalMetadata.Add(NuGetProjectMetadataKeys.UniqueName, uniqueName);
             InternalMetadata.Add(NuGetProjectMetadataKeys.ProjectId, projectId);
 
-            var supportedFrameworks = _project.GetSupportedFrameworksAsync(CancellationToken.None)
-                .Result
+            NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                var supportedFrameworks = (await _project.GetSupportedFrameworksAsync(CancellationToken.None))
```

Comment:
>await _project.GetSupportedFrameworksAsync(CancellationToken.None) [](start = 43, length = 66)

style: using `await` within sub-expressions makes debugging harder, since you can't see the intermediate value returned by the async call. I recommend you assign just this await expression to a local variable, then reference that in the broader context.